### PR TITLE
Show user status in UserScreen, PeopleAutocomplete and right drawer.

### PR DESCRIPTION
**Pull request by vishwesh3**
Use presence to get status.

screen shot 2017-07-19 at 9 19 43 am

### CODE REVIEW

File: src/common/ImageAvatar.js [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
-import { Image } from 'react-native';
+import { Image, StyleSheet } from 'react-native';
 
 import { nullFunction } from '../nullObjects';
 import { Touchable } from './';
+import UserStatusIndicator from '../common/UserStatusIndicator';
+
+const componentStyles = StyleSheet.create({
+  status: {
+    marginLeft: 21,
```
Comment (borisyankov) [inline src/common/ImageAvatar.js]: That does not seem to work universally.
We might have different avatar sizes.
Can you think of a way to make it work for variable sizes?

File: src/common/ImageAvatar.js [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+    marginTop: 21,
+  },
+});
 
 export default class ImageAvatar extends PureComponent {
   props: {
     avatarUrl: string,
     size: number,
     shape: string,
+    status?: string,
```
Comment (borisyankov) [inline src/common/ImageAvatar.js]: The type of it is 'UserStatus' from types.

File: src/common/ImageAvatar.js [hunk 3]
```diff
@@ -32,7 +41,9 @@ export default class ImageAvatar extends PureComponent {
 
     return (
       <Touchable onPress={onPress} style={touchableStyle}>
-        <Image style={imageStyle} source={{ uri: avatarUrl }} resizeMode="contain" />
+        <Image style={imageStyle} source={{ uri: avatarUrl }} resizeMode="contain">
+          {status && <UserStatusIndicator style={componentStyles.status} status={status} />}
```
Comment (borisyankov) [inline src/common/ImageAvatar.js]: Actually, I think we want to keep this in the Avatar.js component. Any reason to prefer moving it here?

File: src/users/userSelectors.js [hunk 4]
```diff
@@ -33,14 +33,24 @@ export const getAllActiveUsers = createSelector(getUsers, allUsers =>
   allUsers.filter(user => user.isActive),
 );
 
+export const getAllActiveUsersWithStatus = createSelector(
+  getAllActiveUsers,
+  getPresence,
+  (allUsers, presence) =>
+    allUsers.map(user => ({
+      ...user,
```
Comment (borisyankov) [inline src/users/userSelectors.js]: That will have very bad performance, though from code expressiveness is good.
On each iteration, you will be cloning the whole object.
Look at `reduce()`

File: src/common/ImageAvatar.js (hunk 1 above)
Comment (vishwesh3) [inline src/common/ImageAvatar.js]: Done 👍 Now there is no margin attribute.

File: src/common/ImageAvatar.js (hunk 2 above)
Comment (vishwesh3) [inline src/common/ImageAvatar.js]: Done 👍

File: src/common/ImageAvatar.js (hunk 3 above)
Comment (vishwesh3) [inline src/common/ImageAvatar.js]: Done 👍

File: src/users/userSelectors.js (hunk 4 above)
Comment (vishwesh3) [inline src/users/userSelectors.js]: Done 👍

File: src/users/userSelectors.js [hunk 5]
```diff
@@ -33,6 +33,17 @@ export const getAllActiveUsers = createSelector(getUsers, allUsers =>
   allUsers.filter(user => user.isActive),
 );
 
+export const getAllActiveUsersWithStatus = createSelector(
+  getAllActiveUsers,
+  getPresence,
+  (allUsers, presence) =>
+    allUsers.reduce((users, user) => {
+      user.status = (presence.find(x => x.email === user.email) || NULL_PRESENCE).status;
```
Comment (borisyankov) [inline src/users/userSelectors.js]: You are mutating the input here. And shouldn't.

Comment (borisyankov) [inline src/users/userSelectors.js]:
```js
users.push({
  ...user,
  status: ...,
});
```

Comment (borisyankov) [inline src/users/userSelectors.js]: Add a `deepFreeze()` call first, to the test. And you will see it falling.

Comment (vishwesh3) [inline src/users/userSelectors.js]: Done 👍

**Comment by borisyankov** [comment]
Looks good.

**Pull request merged**